utils/certs/gen_certificates.sh

```diff
@@ -38,146 +38,210 @@
+    echo "[ signing_admin ]" >> "${CA_HOME}/ca.cnf"
+    echo "keyUsage = critical,digitalSignature,keyEncipherment" >> \
+        "${CA_HOME}/ca.cnf"
+    echo "extendedKeyUsage = clientAuth" >> "${CA_HOME}/ca.cnf"
+    echo "" >> "${CA_HOME}/ca.cnf"
+    echo "[ signing_test ]" >> "${CA_HOME}/ca.cnf"
+    echo "keyUsage = critical,digitalSignature,keyEncipherment" >> \
+        "${CA_HOME}/ca.cnf"
+    echo "extendedKeyUsage = clientAuth" >> "${CA_HOME}/ca.cnf"
 }
 
 function generate_ca_cert () {
-	echo "Generating Private CA Root Certificate"
-	# Generate Private key and set permissions
-	openssl genrsa -out $PRIVATE/daosCA.key 4096
-	chmod 0400 $PRIVATE/daosCA.key
-	# Generate CA Certificate
-	openssl req -new -x509 -config "$CONFIGS/ca.cnf" -days 1095  -sha512 \
-		-key $PRIVATE/daosCA.key \
-		-out $CERTS/daosCA.crt -batch
-	# Reset the the CA index
-	rm -f ./daosCA/index.txt ./daosCA/serial.txt
-	touch ./daosCA/index.txt
-	echo '01' > ./daosCA/serial.txt
-	echo "Private CA Root Certificate created in ./daosCA"
+    echo "Generating Private CA Root Certificate"
+    # Generate Private key and set permissions
+    openssl genrsa -out "${PRIVATE}/daosCA.key" 4096
+    chmod 0400 "${PRIVATE}/daosCA.key"
+    # Generate CA Certificate
+    openssl req -new -x509 -config "$CA_HOME/ca.cnf" -days 1095  -sha512 \
+        -key "${PRIVATE}/daosCA.key" \
+        -out "${CERTS}/daosCA.crt" -batch
+    # Reset the the CA index
+    rm -f "${CA_HOME}/index.txt" "${CA_HOME}/serial.txt"
+    touch "${CA_HOME}/index.txt"
+    echo '01' > "${CA_HOME}/serial.txt"
+    echo "Private CA Root Certificate created in ${CA_HOME}"
 }
 
 function generate_agent_cert () {
-	echo "Generating Agent Certificate"
-	# Generate Private key and set its permissions
-	openssl genrsa -out $CERTS/agent.key 4096
-	chmod 0400 $CERTS/agent.key
-	# Generate a Certificate Signing Request (CRS)
-	openssl req -new -config "$CONFIGS/agent.cnf" -key $CERTS/agent.key \
-		-out agent.csr -batch
-	# Create Certificate from request
-	openssl ca -config "$CONFIGS/ca.cnf" -keyfile $PRIVATE/daosCA.key \
-		-cert $CERTS/daosCA.crt -policy signing_policy \
-		-extensions signing_agent -out $CERTS/agent.crt \
-		-outdir $CERTS -in agent.csr -batch
-
-	echo "Required Agent Certificate Files:
-	$CERTS/daosCA.crt
-	$CERTS/agent.key
-	$CERTS/agent.crt"
+    echo "Generating Agent Certificate"
+    # Generate Private key and set its permissions
+    openssl genrsa -out ${CERTS}/agent.key 4096
+    chmod 0400 ${CERTS}/agent.key
+    # Generate a Certificate Signing Request (CRS)
+    openssl req -new -config "$CONFIGS/agent.cnf" -key ${CERTS}/agent.key \
+        -out ${CA_HOME}/agent.csr -batch
+    # Create Certificate from request
+    openssl ca -config "$CA_HOME/ca.cnf" -keyfile ${PRIVATE}/daosCA.key \
+        -cert ${CERTS}/daosCA.crt -policy signing_policy \
+        -extensions signing_agent -out ${CERTS}/agent.crt \
+        -outdir ${CERTS} -in ${CA_HOME}/agent.csr -batch
```

Comment:
(lint) Double quote to prevent globbing and word splitting. [SC2086]
(lint) Double quote to prevent globbing and word splitting. [SC2086]